lib/src/input_validation/validate_function.dart

```diff
@@ -35,7 +46,7 @@ class ValidateFunction {
     for(int i=0;(i<inputString.length && valid);i++) {
       String token = _sanitizeToken(inputString[i]);
 
-      if(Pattern.validOperand.hasMatch(token) || token.length == 1) {  // numbers or operands
+      if(operands.contains(token) || Pattern.validNumber.hasMatch(token)) {  // numbers or operands
```

Comment:
`validNumber` should be `validOperand`, otherwise pi or decimal numbers will fail.

Reply:
yea i agree with the sentiment here, that regex wasnt quite working. I can look into it